lib/Demangling/Remangler.cpp

```diff
@@ -1528,18 +1545,32 @@ void Remangler::mangleImplFunctionType(Node *node) {
         break;
       }
       case Node::Kind::ImplFunctionAttribute: {
-        char FuncAttr = llvm::StringSwitch<char>(Child->getText())
-                        .Case("@convention(block)", 'B')
-                        .Case("@convention(c)", 'C')
-                        .Case("@convention(method)", 'M')
-                        .Case("@convention(objc_method)", 'O')
-                        .Case("@convention(closure)", 'K')
-                        .Case("@convention(witness_method)", 'W')
-                        .Case("@yield_once", 'A')
-                        .Case("@yield_many", 'G')
-                        .Case("@async", 'H')
-                        .Default(0);
+        StringRef text = Child->hasText()
```

Comment:
Nit: maybe it's just me, but I find nested ternary expressions hard to reason about.

Reply:
Refactored out.